pages/repo/raw.js

```diff
@@ -0,0 +1,237 @@
+import React, { useState, useEffect, useRef } from 'react'
+import Frame from '../../components/Frame'
+import RepoNav from '../../components/Repo/Nav'
+import initLocalStore from '../../lib/utils/localStorage'
+import { withRouter } from 'next/router'
+import withT from '../../lib/withT'
+import withAuthorization from '../../components/Auth/withAuthorization'
+import withMe from '../../lib/withMe'
+import { compose, graphql } from 'react-apollo'
+import { stringify, parse } from '@orbiting/remark-preset'
+import { css } from 'glamor'
+import {
+  A,
+  Button,
+  mediaQueries,
+  colors,
+  fontFamilies
+} from '@project-r/styleguide'
+import { Router } from '../../lib/routes'
+import CircleIcon from 'react-icons/lib/md/lens'
+import gql from 'graphql-tag'
+import { Controlled as CodeMirror } from 'react-codemirror2'
+import {
+  UncommittedChanges,
+  withUncommitedChanges,
+  withUncommittedChangesMutation
+} from '../../components/VersionControl/UncommittedChanges'
+import BranchingNotice from '../../components/VersionControl/BranchingNotice'
+
+const METADATA_SEPARATOR = '---\n\n'
+
+const styles = css({
+  background: colors.secondaryBg,
+  padding: 10,
+  '& .CodeMirror': {
+    height: 'auto',
+    width: 800,
+    margin: 'auto',
+    border: `1px solid ${colors.divider}`,
+    fontFamily: fontFamilies.monospaceRegular,
+    fontSize: 14,
+    color: colors.text,
+    lineHeight: 2
+  },
+  '& pre.CodeMirror-line': {
+    padding: '0 15px'
+  }
+})
+
+const getDocumentContent = gql`
+  query getDocumentContent($repoId: ID!, $commitId: ID!) {
+    repo(id: $repoId) {
+      commit(id: $commitId) {
+        document {
+          content
+        }
+      }
+    }
+  }
+`
+
+// yeah I know that's ugly
+if (typeof window !== 'undefined') {
+  require('../../components/editor/utils/codemirror-md')
+  // TODO: collapse sections -> doesnt work grrrr!
+  require('codemirror/addon/fold/foldcode')
+  require('codemirror/addon/fold/foldgutter')
+  require('codemirror/addon/fold/xml-fold') // we want to fold the sections
+}
+
+const hasOpenSections = md => {
+  if (!md) return
+  return (
+    md.match(/<section>/g).length !== md.match(/<hr \/><\/section>/g).length
+  )
+}
+
+export default compose(
+  withRouter,
+  withT,
+  withAuthorization(['editor']),
+  withMe,
+  graphql(getDocumentContent, {
+    skip: ({ router }) =>
+      router.query.commitId === 'new' || !router.query.commitId,
+    options: ({ router }) => ({
+      variables: {
+        repoId: router.query.repoId,
+        commitId: router.query.commitId
+      }
+    })
+  }),
+  withUncommitedChanges({
+    options: ({ router }) => ({
+      variables: {
+        repoId: router.query.repoId
+      }
+    })
+  }),
+  withUncommittedChangesMutation
+)(({ t, router, data, uncommittedChanges, hasUncommitedChanges }) => {
+  const { repoId, commitId } = router.query
+  const [store, setStore] = useState(undefined)
+  const storeRef = useRef()
+  storeRef.current = store
```

Comment:
This ref seem unnecessary. Are you sure it's needed?

Reply:
Yes it is. The `onSave() ` function cannot access the store otherwise.

Reply:
are you sure that that is actually the case? I don't see any reason why that would be—the `store` variable should be accessible from the scope of `onSave`—and just doing `store.set('editorState', mdast)` in `onSave` works on my machine.

Reply:
mmmh - I'll double-check then